resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/k8s/features/HadoopConfExecutorFeatureStep.scala

```diff
@@ -0,0 +1,51 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.spark.deploy.k8s.features
+
+import io.fabric8.kubernetes.api.model.HasMetadata
+
+import org.apache.spark.deploy.k8s.{KubernetesConf, KubernetesExecutorSpecificConf, SparkPod}
+import org.apache.spark.deploy.k8s.Constants._
+import org.apache.spark.deploy.k8s.features.hadooputils.HadoopBootstrapUtil
+import org.apache.spark.internal.Logging
+
+ /**
+  * This step is responsible for bootstraping the container with ConfigMaps
+  * containing Hadoop config files mounted as volumes and an ENV variable
+  * pointed to the mounted file directory.
+  */
+private[spark] class HadoopConfExecutorFeatureStep(
+  kubernetesConf: KubernetesConf[KubernetesExecutorSpecificConf])
+  extends KubernetesFeatureConfigStep with Logging{
+
+  override def configurePod(pod: SparkPod): SparkPod = {
+     val sparkConf = kubernetesConf.sparkConf
+     val maybeHadoopConfDir = sparkConf.getOption(HADOOP_CONF_DIR_LOC)
+     val maybeHadoopConfigMap = sparkConf.getOption(HADOOP_CONFIG_MAP_NAME)
+     require(maybeHadoopConfDir.isDefined && maybeHadoopConfigMap.isDefined,
```

Comment:
Why `&&` here?

Reply:
This is a check to make sure the executor has the ConfigMap defined as well as the HADOOP_CONF_DIR.

Reply:
But checking the presence of `HADOOP_CONF_DIR_LOC` is not really checking that the env variable `HADOOP_CONF_DIR` is set, right?

Reply:
`HADOOP_CONF_DIR_LOC -> kubernetesConf.hadoopConfDir.get` so it is by extension.

Reply:
I thought users either use `HADOOP_CONF_DIR` to specify the path of local Hadoop config, or they specify an existing ConfigMap storing the Hadoop config. So it should be one or the other, right? BTW: I don't see option to allow users specify an existing ConfigMap for Hadoop config.

Reply:
ATM, it is just HADOOP_CONF_DIR; the ability to use an existing ConfigMap is an additional feature that will only be relevant with the pod-template feature, which is still unmerged. As such, I didn't include that check yet. It was just a mentioned in the docs.